### ISSUE TITLE
Assign eligible role to management group

### ISSUE THREAD

**Issue by pimvd** [Azure/azure-cli#30332]
**Related command**
[az role assignment create](https://learn.microsoft.com/en-us/cli/azure/role/assignment?view=azure-cli-latest#az-role-assignment-create)

**Is your feature request related to a problem? Please describe.**
In Azure Portal I am assigning roles to a Management Group. I want to create an Eligible role assignment (which is recommended) via Azure CLI. This is currently in preview in the Azure Portal and not available in the Azure CLI. I would like to create the eligible role via Azure CLI.

**Describe the solution you'd like**
I would like to be able to choose a **Assignment type**, **Assignment duration**, **Start date and time** and **End date and time** or **expirationDuration** (e.g. 1 year, ISO 8601 https://en.wikipedia.org/wiki/ISO_8601#Durations ) like in the Azure Portal. 
Example: `az role assignment create --assignee "<Group GUID>" --role "Contributor" --scope "/providers/Microsoft.Management/managementGroups/<MANAGEMENT GROUP ID>" --assignment-type "Eligible" --assignment-duration "TimeBound" --expiration-duration "P1Y"
`

**Describe alternatives you've considered**
None

**Additional context**
Basically I want to be able to use the recommended assignment type, Eligible, via Azure CLI.
![Image](https://github.com/user-attachments/assets/add48d8d-bd8f-4ec1-8aa7-ba96cd0ca2d9)

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by discapes** [comment]
I find this to be an important feature and would really encourage creating Eligible type assignments instead of permanent.
